zdm/grid.py

```diff
@@ -137,12 +145,64 @@ def load_grid(self, gridfile, zfile, dmfile):
         self.dmvals = io.load_data(dmfile)
         self.check_grid()
         self.volume_grid()
-
-    def check_grid(self):
-
+    
+    def get_dm_coeffs(self, DMlist):
+        """
+        Returns indices and coefficients for interpolating between DM values
+        
+        dmlist: np.ndarray of dispersion measures (extragalactic!)
+        """
+        # get indices in dm space
+        kdms=DMlist/self.ddm - 0.5 # idea: if DMobs = ddm, we are half-way between bin 0 and bin 1
+        Bin0 = np.where(kdms < 0.)[0] # if DMs are in the lower half of the lowest bin, use lowest bin only
```

Comment:
Not sure I follow this comment. Is this line not redundant as kdms should start at 1 so -0.5 means it will never go negative? Same for kzs below.